src/AEG/AEGUtils.ts

```diff
@@ -154,22 +172,23 @@ function edgesWithin(rect1: Rectangle, rect2: Rectangle): boolean {
 export function pointInRect(rect: Rectangle, point: Point): boolean {
     const rectCorners = rect.getCorners();
 
-    //Points on edges are considered to be contained
+    //Points on edges are considered to be intersected, not contained
     return (
-        rectCorners[0].x <= point.x &&
-        rectCorners[1].x >= point.x &&
-        rectCorners[0].y <= point.y &&
-        rectCorners[2].y >= point.y
+        rectCorners[0].x < point.x &&
+        rectCorners[1].x > point.x &&
+        rectCorners[0].y < point.y &&
+        rectCorners[2].y > point.y
     );
 }
 
 /**
- * Method that checks whether a point is within the given ellipse.
+ * Method that returns the value of a point compared to the ellipse
  * @param ellipse The given ellipse
  * @param otherPoint The point that might be inside the given ellipse.
- * @returns True, if the point is inside the given ellipse. Else, false
+ * @returns 1, if the point is on the ellipse.
+ * Returns <1 if the point is completely within the ellipse.
```

Comment:
You're missing a case; this can also be greater than 1 if the point is outside the ellipse.